src/buffer.rs

```diff
@@ -789,7 +779,7 @@ mod tests {
     fn test_buffer_to_string_ignore_newline() {
         let mut buf = Buffer::new();
         buf.insert(0, ['h', 'e', '\\', '\n', 'l', 'l', 'o'].iter());
-        assert_eq!("hello".to_owned(), String::from(buf));
+        assert_eq!("he\\\nllo".to_owned(), String::from(buf));
```

Comment:
Why did this test change?  Would not a backslash at the end of a line still "consume" the newline?

Reply:
I removed this test. It doesn't make sense anymore. It's just testing this method

```

impl From<Buffer> for String {
    fn from(buf: Buffer) -> Self {
        buf.data
    }
}
```

and that method used to strip newlines.

We decided this was bad because it meant that autocompletion did not include newlines, which was confusing.  As it stands now that test technically showed us the old code doesn't exist. 

The other thing that's confusing is the buffer doesn't, and I don't know if it should or shouldn't, understand anything about multiple lines. It depends on future development. Going forward, when doing multi line editing we want up and down to behave the way you'd expect in the terminal and currently they do not. But I don't know if we are going to make buffer multiline aware, and have the keymap implementations "deal" with that. or keep buffer stupid, where it just holds a string, and that string might happen to have some newlines in it, that the keymap will have to work around. or maybe there will be another layer in-between the two, that will take care of that. I don't know yet.